source/fuzz/transformation_add_type_array.cpp

```diff
@@ -39,9 +39,11 @@ bool TransformationAddTypeArray::IsApplicable(
   }
   auto element_type =
       ir_context->get_type_mgr()->GetType(message_.element_type_id());
-  if (!element_type || element_type->AsFunction()) {
-    // The element type id either does not refer to a type, or refers to a
-    // function type; both are illegal.
+  if (!element_type || element_type->AsFunction() ||
+      fuzzerutil::HasBlockOrBufferBlockDecoration(ir_context,
+                                                  message_.element_type_id())) {
```

Comment:
Probably worth mentioning this in the docs.

Reply:
How strange - in my working copy (which appears to be committed) the comment is already updated.

Reply:
Oh, wait - I guess you mean the top level documentation for the transformation. Good idea - will do that.